apf/apfMatrix.h

```diff
@@ -12,6 +12,20 @@
   \brief The APF linear algebra matrix interface */
 
 #include "apfVector.h"
+#include <limits> // epsilon value
+
+// this function doesn't use the "best" method for floating point comparison
+// FIXME update this with something like:
+// https://github.com/google/googletest/blob/dccc2d67547a5a3a97e4f211f39df931c6fbd5d5/googletest/include/gtest/internal/gtest-internal.h#L334-L347
+// the implemented expression comes from the art of computer programming by Knuth
```

Comment:
Can you provide a more detailed reference?  Which volume chapter etc.. did it come from?  https://en.wikipedia.org/wiki/The_Art_of_Computer_Programming

Reply:
This isn't addressed.